docs/architecture/0002-import-subcommand.md

```diff
@@ -0,0 +1,68 @@
+<!--
+This is a template for [Documenting Architecture Decisions - Michael Nygard](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions).
+
+You can use [adr-tools](https://github.com/npryce/adr-tools) for managing the ADR files.
+
+In each ADR file, write the following sections.
+-->
+# 2. Import subcommand
+
+Date: 2024-07-09
+
+## Status
+<!--
+A decision may be "proposed" if the project stakeholders haven't agreed with it yet, or "accepted" once it is agreed.
+If a later ADR changes or reverses a decision, it may be marked as "deprecated" or "superseded" with a reference to
+its replacement.
+-->
+Proposed
+
+## Context
+<!--
+This section describes the forces at play, including technological, political, social, and project local. These forces
+are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply
+describing facts.
+-->
+There is currently no capability of importing SBOM or protobom data from either `stdin` or local filesystem path(s).
+
+## Decision
+<!--
+This section describes our response to these forces. It is stated in full sentences, with active voice. "We will …"
+-->
+Introduce an `import` command that will accept one of the following options as input:
+
+- stream of bytes piped from `stdin`
+- path to a local file or files as optional positional arguments
+  - alternatively, input files could could be specified with an explicit flag, such as `--input`, `--file`, `--path`, etc.
+
+The supported input types will be:
+
+- CycloneDX SBOM
+- SPDX SBOM
+- Protobom `Document` (such as the serialized protocol buffer storage format used by the Protobom `FileSystemBackend`)
+
+The `fetch` command will now be able to simply fetch raw bytes data and leverage the new `import` logic to store.
+
+## Consequences
+<!--
```

Comment:
I've had some thoughts about how to handle multiple piped SBOMs in general, and I'm not sure how common of a use case that might be. I also don't know of any tools off the top of my head that would be providing multiple SBOMs as output.

One option might be to use a special delimiter to signify the end of a document and start of another one. Another might be to accept multiple SBOMs if they're wrapped in a JSON list.

I think it makes sense for now to handle it by only accepting a single root SBOM via the pipeline, and having a flag to attempt to import external references recursively, but definitely open to other ideas on that.

Reply:
Recursive importing would have some interesting edge cases. I am assuming you can reference multiple SBOMs and if some worked and some did not that might require some sort of behavior. Should we throw warnings and have the user decide explicitly if they want to accept the partial recursive import?

Reply:
I think the concept of `import` makes the most sense for dealing exclusively with data that is already available on `stdin` or the local filesystem. To me, resolving and fetching external references seems like it should be provided by the `fetch` command.

I could see a use case where a user might want to do a batch import, in which case it might make sense to also allow `.tar` files and OCI archives as input. It would then logically follow that `export` should be able to output the same file types. What are your thoughts?

@jhoward-lm @eddiezane @mfrystacky @ashearin